tools/minidumpserver.py

```diff
@@ -0,0 +1,530 @@
+            socket.send(b"+")
+
+            return data
+        else:
+            # NACK
+            logger.debug(f"NACK (checksum {in_chksum} != {checksum}")
+            socket.send(b"-")
+
+            return None
+
+    def put_gdb_packet(self, data):
+        socket = self.socket
+        if socket is None:
+            return
+
+        checksum = 0
+        for d in data:
+            checksum += d
+
+        pkt = b"$" + data + b"#"
+
+        checksum = checksum % 256
+        pkt += format(checksum, "02X").encode()
+
+        logger.debug(f"Sending GDB packet: {pkt}")
+
+        socket.send(pkt)
+
+    def handle_signal_query_packet(self):
+        # the '?' packet
+        pkt = b"S"
+        pkt += format(self.gdb_signal, "02X").encode()
+
+        self.put_gdb_packet(pkt)
+
+    def handle_register_group_read_packet(self):
+        reg_fmt = "<I"
+        pkt = b""
+
+        for reg in self.logfile.registers:
+            if reg != b"x":
+                bval = struct.pack(reg_fmt, reg)
+                pkt += binascii.hexlify(bval)
+            else:
+                # Register not in coredump -> unknown value
+                # Send in "xxxxxxxx"
+                pkt += b"x" * 8
+
+        self.put_gdb_packet(pkt)
+
+    def handle_register_single_read_packet(self, pkt):
+        # Mark registers as "<unavailable>".
+        # 'p' packets are usually used for registers
+        # other than the general ones (e.g. eax, ebx)
+        # so we can safely reply "xxxxxxxx" here.
+        self.put_gdb_packet(b"x" * 8)
+
+    def handle_register_group_write_packet(self):
+        # the 'G' packet for writing to a group of registers
+        #
+        # We don't support writing so return error
+        self.put_gdb_packet(b"E01")
+
+    def handle_register_single_write_packet(self, pkt):
+        # the 'P' packet for writing to registers
+        #
+        # We don't support writing so return error
+        self.put_gdb_packet(b"E01")
+
+    def handle_memory_read_packet(self, pkt):
+        # the 'm' packet for reading memory: m<addr>,<len>
+
+        def get_mem_region(addr):
+            for r in self.mem_regions:
+                if r["start"] <= addr <= r["end"]:
+                    return r
+
+            return None
+
+        # extract address and length from packet
+        # and convert them into usable integer values
+        addr, length = pkt[1:].split(b",")
+        s_addr = int(b"0x" + addr, 16)
+        length = int(b"0x" + length, 16)
+
+        # FIXME: Need more efficient way of extracting memory content
+        remaining = length
+        addr = s_addr
+        barray = b""
+        r = get_mem_region(addr)
+        while remaining > 0:
+            if r is None:
+                barray = None
+                break
+
+            if addr > r["end"]:
+                r = get_mem_region(addr)
+                continue
+
+            offset = addr - r["start"]
+            barray += r["data"][offset : offset + 1]
```

Comment:
is it possible to
```suggestion
            barray += r["data"][offset:offset + 1]
```
to get LINT happy?

Reply:
Lint Ci have double standard about `:`

Reply:
maybe `()` can solve the issue? `r["data"][offset:(offset + 1)]`?